split sample and patient files

aderidder opened the pull request:
fix #1 
given the huge variety available, I've gone for a solution which uses a defaultAttributes.txt file. My idea is that we try to create a super-awesome defaultAttributes file, which should hopefully lead to few collisions in cbioportal. Users can also add a custom attributes file; in case of collisions the default is used (or cbio will complain).
The attributes file works as follows:
attributename<tab>B/P/S
where P is patientfile, S is samplefile and B is both

seems to work for my decode wp3, I'll also try decode wp5 when I have time

Commit:
create clinical patient file and clinical sample file based on default attributes as defined in defaultAttributes.txt
also added support for a custom attributes file

j-hudecek commented:
Thanks a lot this looks great. Since you're adding a new configuration element, could you please add it to transmart2cbio.mapping.txt (that should be a mapping file template with all possible config attributes) and a sample (super simple) attribute def file?

Commit:
when the attribute types file is read, spaces are now replaced with underscores
defaultAttributeTypes is now a more simple example file
added new configuration element to transmart2cbio.mapping.txt

aderidder commented:
Hi I changed the deafaultAttributeTypes file, which is now basically a super simple attribute def file. 
I also discovered a stupid mistake which I fixed as well.
